src/safeds/data/tabular/containers/_table.py

```diff
@@ -328,6 +336,10 @@ def __eq__(self, other: Any) -> bool:
             return NotImplemented
         if self is other:
             return True
+        if self.number_of_rows == 0 and other.number_of_rows == 0 and self.column_names == other.column_names:
+            return True
```

Comment:
The sorting stuff was already done previous to our commit, that way tables with columns (a, b) and (b, a) are considered equal. This behavior is not meant to be changed by the check for an empty table.

Reply:
The behavior is not changed with my suggestion. It is only necessary if you want to compare the pandas DataFrames. If you have no rows in your table this is unnecessary and you can just return weather the column names are equal or not.

Reply:
Ok I see your point. With your version there is a problem if the columns have different types: 

```py
from safeds.data.tabular.containers import Table

table = Table({"Column1": [1], "Column2": [None]})
table2 = Table({"Column2": [], "Column1": []})
table = table.remove_rows_with_missing_values()

print(table == table2) # Should be True but is False
```

See my suggestion in https://github.com/Safe-DS/Stdlib/pull/314#discussion_r1204556867 which should solve both issues.

Reply:
I do not quite get why this should be 'True'. It is false indeed as one of those tables has data points, the other one does not.
Additionally, I see that your proposed change would help that, but that is because it changes the behavior of the __eq__ function into ignoring data points and data types entirely, which is not the way the equality check is supposed to work...

Reply:
The scheme checkup together with this data checkup would have flagged this false, I think. However, I do agree that your proposed solution is of far better style, so I already committed it before I could test my (now irrelevant) guess on the original behavior😅